"""
The ``100-matrix_mul`` module
"""

importing ``matrix_mul` from the module

     >>> matrix_mul = __import__("100-matrix_mul").matrix_mul

def matrix_mul(m_a, m_b):
""" A function that multiplies two matrices.

    Args: 
        m_a: First matrix, m_b: Second Matrix
    Return: Returns a new matrix """


Tests when wrong Type is passed
::
        >>> matrix_mul([[2, 3, 4], [4, 2, 3]], None)
	Traceback (most recent call last):
	...
	TypeError: m_b must be a list

	>>> matrix_mul("school", [[2, 3, 4], [4, 2, 3]])
	Traceback (most recent call last):
	...
	TypeError: m_a must be a list

	>>> matrix_mul([{"USA":"Washington D.C."}, {"France":"Paris", "India":"New Delhi"}], [[1, 2, 3], [2, 3, 4]])
	Traceback (most recent call last):
	...
	TypeError: m_a should contain only integers or floats

	>>> matrix_mul([2, 3], [2, 3])
	Traceback (most recent call last):
	...
	TypeError: m_b must be a list of lists

	>>> matrix_mul([[1, 2], [2, 3]], [[1, 3], (4, 4)])
	Traceback (most recent call last):
	...
	TypeError: m_b must be a list of lists

    >>> matrix_mul([[1.3, 2, 4], [2.5, 3], [3, 4]], [[1, 2], [4, 5]])
	Traceback (most recent call last):
	...
	TypeError: each row of m_a must be of the same size


    Tests when wrong values are passed as arguement
::
	>>> matrix_mul([[]], [[2, 4]])
	Traceback (most recent call last):
	...
	ValueError: m_a can't be empty

	>>> matrix_mul([[1, 4.2], [1, 4.0]], [])
	Traceback (most recent call last):
	...
	ValueError: m_b can't be empty

    >>> matrix_mul([[4, 3], [1, 4], [3, 5]], [3, 1, 4])
	Traceback (most recent call last):
	...
	ValueError: m_a and m_b can't be multiplied

    >>> matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]])
    [[7, 10], [15, 22]]

    Checking for one too few args:
	>>> matrix_mul([[1, 2], [3, 4]])
	Traceback (most recent call last):
	...
	TypeError: matrix_mul() missing 1 required positional argument: 'm_b'

Checking for too many args:
	 >>> matrix_mul([[1, 2], [3, 4]], [[5, 6], [7, 8]], [[9, 10], [11, 12]])
	 Traceback (most recent call last):
	 ...
	 TypeError: matrix_mul() takes 2 positional arguments but 3 were given

Checking for when no argument is passsed
    >>> matrix_mul()
    Traceback (most recent call last):
	 ...
	TypeError: matrix_mul() missing 2 required positional arguments: 'm_a' and 'm_b'